pkg/rancher/rancher.go

```diff
@@ -104,7 +104,13 @@ func New(ctx context.Context, clientConfg clientcmd.ClientConfig, opts *Options)
 	}
 
 	if features.MCM.Enabled() && !features.Fleet.Enabled() {
-		return nil, fmt.Errorf("multi-cluster-management features requires feature fleet=true")
+		logrus.Info("fleet can't be turned off when MCM is enabled. Turning on fleet and turning off continuous-delivery")
+		if err := features.SetFeature(wranglerContext.Mgmt.Feature(), features.Fleet, true); err != nil {
+			return nil, err
+		}
+		if err := features.SetFeature(wranglerContext.Mgmt.Feature(), features.Gitops, false); err != nil {
```

Comment:
Is it necessary that we assume the desired value and set this on their behalf? If we remove the fleet feature flag we can just say that it has been removed and a new feature flag for gitops has been added which the user can disable if they do not desire it.

Reply:
Same, can't remove fleet feature since downstream cluster will need to set it to false.